File: payroll-frontend/src/types/employee.ts
export interface PageResponse<T> {
  data: T[];
  total: number;
  page: number;
  limit: number;
}

export interface Department {
  id: number;
  code: string;
  name: string;
  description: string | null;
  parentId: number | null;
  parentName: string | null;
}

export interface Position {
  id: number;
  code: string;
  name: string;
  description: string | null;
  departmentId: number;
  departmentName: string;
  salaryGrade: number | null;
}

export interface Employee {
  id: number;
  employeeId: string;
  firstName: string;
  lastName: string;
  fullName: string;
  dateOfBirth: string | null;
  gender: string | null;
  hireDate: string;
  departmentId: number;
  departmentName: string;
  positionId: number | null;
  positionName: string | null;
  categoryId: number;
  categoryCode: string;
  echelon: number | null;
  baseSalary: number | null;
  familyStatus: string | null;
  numberOfChildren: number | null;
  nationalId: string | null;
  cnssNumber: string | null;
  phoneNumber: string | null;
  email: string | null;
  address: string | null;
  city: string | null;
  zipCode: string | null;
  paymentMethod: string | null;
  bankAccountNumber: string | null;
  bankCode: string | null;
  employmentStatus: string;
  terminationDate: string | null;
  terminationReason: string | null;
}

/** Shape of the create/edit form payload sent to the API. */
export interface EmployeeFormValues {
  employeeId?: string;
  firstName: string;
  lastName: string;
  hireDate: string;
  departmentId: number | '';
  positionId?: number | '';
  categoryId: number | '';
  echelon?: number | '';
  gender?: string;
  email?: string;
  cnssNumber?: string;
  baseSalary?: number | '';
  familyStatus?: string;
  numberOfChildren?: number | '';
  phoneNumber?: string;
  employmentStatus?: string;
}

export interface DepartmentRequest {
  code: string;
  name: string;
  description?: string;
  parentId?: number | null;
}

export const EMPLOYMENT_STATUSES = ['ACTIVE', 'INACTIVE', 'LEAVE', 'TERMINATED'] as const;

/** Tunisian IRPP family-status codes: C = single, M0–M4 = married with 0–4 dependants. */
export const FAMILY_STATUSES = [
  { code: 'C', label: 'C — Single' },
  { code: 'M0', label: 'M0 — Married, 0 dependants' },
  { code: 'M1', label: 'M1 — Married, 1 dependant' },
  { code: 'M2', label: 'M2 — Married, 2 dependants' },
  { code: 'M3', label: 'M3 — Married, 3 dependants' },
  { code: 'M4', label: 'M4 — Married, 4 dependants' },
] as const;

File: payroll-backend/src/main/java/com/maram/payroll/employee/entity/Employee.java
package com.maram.payroll.employee.entity;

import com.maram.payroll.common.domain.Auditable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Employee master record. The single source of truth for everything downstream
 * (attendance, payroll, reporting).
 */
@Entity
@Table(name = "employees")
@Getter
@Setter
@NoArgsConstructor
public class Employee extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false, unique = true)
    private String employeeId;

    @Column(name = "first_name", nullable = false)
    private String firstName;

    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(name = "gender", length = 1)
    private String gender;

    @Column(name = "hire_date", nullable = false)
    private LocalDate hireDate;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "position_id")
    private Position position;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    private SalaryCategory category;

    private Integer echelon;

    @Column(name = "base_salary")
    private BigDecimal baseSalary;

    @Column(name = "hourly_rate")
    private BigDecimal hourlyRate;

    @Column(name = "family_status")
    private String familyStatus;

    @Column(name = "number_of_children")
    private Integer numberOfChildren;

    @Column(name = "national_id")
    private String nationalId;

    @Column(name = "cnss_number")
    private String cnssNumber;

    @Column(name = "phone_number")
    private String phoneNumber;

    private String email;
    private String address;
    private String city;

    @Column(name = "zip_code")
    private String zipCode;

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "bank_account_number")
    private String bankAccountNumber;

    @Column(name = "bank_code")
    private String bankCode;

    @Column(name = "employment_status")
    private String employmentStatus = "ACTIVE";

    @Column(name = "termination_date")
    private LocalDate terminationDate;

    @Column(name = "termination_reason")
    private String terminationReason;
}
